common/src/ticker.rs

```diff
@@ -0,0 +1,35 @@
+use crate::Step;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TickerEntity {
+    pub last_ticked: Step,
+    pub ticker: Ticker,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Ticker {
+    on: bool,
+}
+
+impl Ticker {
+    pub fn new() -> Self {
+        Self { on: false }
+    }
+    pub fn tick(&mut self, _step: Step) {
```

Comment:
A method on `Ticker` _seems_ like the best place to put logic for entities with the `Ticker` component. However, I worry that this comes from an object-oriented way of thinking. With ECS, components are usually thought of as "plain-old data", so the logic that manipulates the `Ticker` component would want to live outside it.

This logic could live directly in `server::sim`, or it could live in a separate module (much like `common::character_controller`, although that is somewhat special because both the server and the client have to run its logic).

It's likely that the `self.on = !self.on` line of code would live in the same function as the `for` loop that loops through all the tickers.

Reply:
You're right, it's very C++ pilled. Will do.